src/components/Header/index.jsx

```diff
@@ -0,0 +1,81 @@
+// react libraries
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+// scss
+import './Header.scss';
+
+/**
+ * @exports
+ * @class Header
+ * @extends Component
+ * @classdesc Creates Header Component
+ *
+ * @returns {JSX} Header Component
+ */
+export class Header extends Component {
+    static propTypes = {
+      history: PropTypes.instanceOf(Object),
+    };
+
+    static defaultProps = {
+      history: {}
+    }
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        searchValue: '',
+      };
+    }
+
+    componentDidMount() {}
+
+  handleInputChange = ({ target: { value } }) => {
+    this.setState({ searchValue: value });
+  };
+
+  handleSearch = ({ keyCode }) => {
+    const { searchValue } = this.state;
+    const { history } = this.props;
+
+    if (keyCode === 13) {
+      history.push(`/search?query=${searchValue}`);
+    }
+  };
+
+  render() {
+    const { searchValue } = this.state;
+    return (
+      <header className="header">
+        <a href="/#">
+          <img src="./src/assets/images/1KbIdeas.svg" alt="1kbIdeas logo" />
+        </a>
+        <div>
+          <input
+            type="text"
+            className="search"
+            placeholder="Search for articles"
+            onKeyUp={this.handleSearch}
+            onChange={this.handleInputChange}
+            value={searchValue}
+          />
+        </div>
+        <nav>
+          <a
+            href="/#"
+            className="header__homeNavLinks header__homeNavLinks--right-margin"
```

Comment:
These class names don't look okay
```suggestion
            className="header__navlink header__navlink--right"
```